src/types/dataBuffer.ts

```diff
@@ -59,7 +59,10 @@ export class DataBuffer {
     
     // Get the underlying ArrayBuffer object.
     get buffer(): ArrayBuffer {
-        return this.array.buffer;
+        // ES2017 has SharedArrayBuffer, which causes TypeScript
+        // compilation errors, since TypedArray.buffer can be either
+        // ArrayBuffer or SharedArrayBuffer
+        return this.array.buffer as ArrayBuffer;
```

Comment:
Good foresight, though this type is not used and will probably be removed. I wrote this before I was aware that webpack would change code using the node `Buffer` type into something browser-compatible by itself.

Though after reading a little more, maybe it would be wise to switch over anyway. Hm. What do you think?